src/interceptors/ClientRequest/utils/recordRawHeaders.test.ts

```diff
@@ -126,6 +126,28 @@ it('records raw headers (Reqest / Request as init)', () => {
   expect(getRawFetchHeaders(request.headers)).toEqual([['X-My-Header', '1']])
 })
 
+it("doesn't change the init headers instanceof (Request / Request as init)", () => {
```

Comment:
Maybe let's rephrase it to "preserves the original headers instance"?

Reply:
Well it's not technically the same header instance...however if you think it's important we could rework this whole thing by pushing looping over the raw headers and set the  on the original instance